classes/migration/SubmissionFilesMigration.inc.php

```diff
@@ -60,7 +60,7 @@ public function up() {
 			$table->bigInteger('submission_file_id')->unsigned();
 			$table->bigInteger('file_id')->unsigned();
 			$table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files');
-			$table->foreign('file_id')->references('file_id')->on('files');
+			$table->foreign('file_id')->references('file_id')->on('files')->onDelete('cascade')->onUpdate('cascade');
```

Comment:
We aren't using foreign key constraints anywhere in the 3.3.0-x codebase -- and the DBMS might not support it. We also don't introduce database changes between builds on a stable branch. You'll have to find a way to manually implement the cascading delete, I'm afraid!

Reply:
Noted. I'll make the necessary updates

Reply:
I've updated the code. Let me know if the approach taken makes sense. 
What I did was to just delete the dependent data within the `PKPFileService::delete` method before deleting the file itself